setup.py

```diff
@@ -12,6 +12,7 @@
     install_requires=[
         "opentelemetry-api==0.15b0",
         "opentelemetry-sdk==0.15b0",
-        "requests~=2.25"
+        "requests~=2.25", 
+        "regex~=2020.11.13"
```

Comment:
Is this only to get access to the `\p{C}` control character class? I am somewhat against adding a dependency for that reason. 

Perhaps we could use the `unicodedata` library which is in the python std lib

```python
import unicodedata
def remove_control_characters(s):
	# iterate over all characters, replacing control characters with _
    s = "".join(ch if unicodedata.category(ch)[0]!="C" else "_" for ch in s)
	# remove leading _
	s = re.sub(r"^_*", "", s)
	# remove trailing _
	s = re.sub(r"_*$", "", s)
	return s
```